### PR TITLE
Capture zypper output + normalize ltp ctcs2 logs before submit to qadb2

### PR DESCRIPTION
**Pull request by mimi1vx**

### CODE REVIEW

File: tests/qam-kgraft/update_kgraft.pm [hunk 1]
```diff
@@ -81,6 +81,8 @@ sub run() {
     # check if automounter works
     check_automounter;
 
+    script_run("while pgrep packagekitd; do pkcon quit; sleep 1; done");
```
Comment (coolo) [inline tests/qam-kgraft/update_kgraft.pm]: not a cool idea. pkcon quit is actually calling packagekitd through dbus - which auto activates packagekitd. So you might have seen the packagekitd's last moment of quitting and then autoactivate it . Saver is actually to call pkcon quit and then while pgrep; sleep; done

File: lib/qam.pm [hunk 2]
```diff
@@ -59,7 +59,26 @@ sub snap_revert {
     my ($svirt, $vm_name, $snapshot) = @_;
     my $ret = $svirt->run_cmd("virsh snapshot-revert $vm_name $snapshot --running");
     die "Snapshot revert $snapshot failed" if $ret;
+}
+
+sub zypper_call_log {
```
Comment (coolo) [inline lib/qam.pm]: this is a full copy of the zypper_call function with a minor difference.  Please integrate this feature into the zypper_call function, possibly by using named parameters

File: tests/qam-kgraft/update_kgraft.pm (hunk 1 above)
Comment (okurz) [inline tests/qam-kgraft/update_kgraft.pm]: or do what we do in most other occasions, e.g. in `tests/console/consoletest_setup.pm`:

```perl
# Stop packagekit
script_run "systemctl mask packagekit.service";
script_run "systemctl stop packagekit.service";
```

or maybe even better to also fix some other potential issues which migh show up later: Just call `consoletest_setup.pm` in before (and `…finish.pm` afterwards).

Comment (coolo) [inline tests/qam-kgraft/update_kgraft.pm]: we don't want to mess with the config of kgraft instances

File: tests/qam-minimal/install_patterns.pm [hunk 3]
```diff
@@ -19,14 +19,14 @@ use testapi;
 sub run {
     select_console 'root-console';
 
-    script_run("while pgrep packagekitd; do pkcon quit; sleep 1; done");
+    script_run("pkcon quit; while pgrep packagekitd; do sleep 1; done");
```
Comment (coolo) [inline tests/qam-minimal/install_patterns.pm]: now that you found out so you have so many duplicates of this - how about a function ? :)

File: lib/utils.pm [hunk 4]
```diff
@@ -275,13 +275,24 @@ sub get_netboot_mirror {
 
 sub zypper_call {
     my $command          = shift;
-    my $allow_exit_codes = shift || [0];
-    my $timeout          = shift || 700;
-    my $str              = hashed_string("ZN$command");
+    my %args             = @_;
+    my $allow_exit_codes = $args{'exitcode'} || [0];
```
Comment (coolo) [inline lib/utils.pm]: you don't need quotes for hash access if the key is a simple string

Comment (mimi1vx) [inline lib/utils.pm]: ok

File: tests/qam-minimal/install_patterns.pm (hunk 3 above)
Comment (mimi1vx) [inline tests/qam-minimal/install_patterns.pm]: good idea , to utils.pm ?

Comment (coolo) [inline tests/qam-minimal/install_patterns.pm]: yes, will do

**Pull request merged**